### PR TITLE
completion: improve docs for activation

### PR DESCRIPTION
**Pull request by senekor**
motivated by [this discussion](https://github.com/martinvonz/jj/pull/4737#discussion_r1839402140)

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: docs/install-and-setup.md [hunk 1]
```diff
@@ -210,12 +210,13 @@ To set up command-line completion, source the output of
...
-like bookmark names as well. You can activate them with the alternative "dynamic"
-instructions below. Please let us know if you encounter any issues, so we can
-ensure a smooth transition once we default to these new completions. If you
-have ideas about specific completions that could be added, please share them
-[here](https://github.com/martinvonz/jj/issues/4763).
+Improved completions are currently in the works, these will complete things like
+bookmark names as well. You can activate them with the alternative "dynamic"
+instructions below. They should still complete everything the static completions
+did, so only activate one of them. Please let us know if you encounter any
+issues, so we can ensure a smooth transition once we default to these new
```
Comment (martinvonz) [inline docs/install-and-setup.md:217]: When should I (as a user) prefer to use the static completions?

Comment (senekor) [reply docs/install-and-setup.md:217]: - You don't care about dynamic completions, so you have no desire to be beta-tester for them.
- You already encountered some issue with the dynamic ones, so you switch back to fix your workflow. (hopefully you also report the issue upstream)

Comment (senekor) [reply docs/install-and-setup.md:217]: I have obviously been using the dynamic completions myself the last couple of days and I haven't experienced any regressions with respect to the static ones. (using fish)

Do you think we should advertise the dynamic completions as _default_, while still providing the static ones as fallback if people run into issues? That would probably lead to more people trying it out.

Comment (martinvonz) [reply docs/install-and-setup.md:217]: > Do you think we should advertise the dynamic completions as _default_, while still providing the static ones as fallback if people run into issues? That would probably lead to more people trying it out.

Yes, that's what I was thinking, but Yuya [reported](https://github.com/martinvonz/jj/pull/4737#discussion_r1839702085) a bug in the Zsh version already, so we probably should not do that yet.

**Review by martinvonz — APPROVED** [review]

**Pull request merged**